### PR TITLE
Changing mail and password dev

### LINKED ISSUE #5315: Changing user's password and email at the same time only successfully changes email
## Current behavior
If the user changes both their email and password for their account, they are logged out (does not occur if only 1 of password or email is changed). When logging in again with the changed email/password combo, the new email is sticking, but the new password is not. Both the password and email change API requests go through without a hitch.

## Expected behavior
The new email AND password values would be changed upon save.

## Steps to reproduce the issue
Login.
Go to the my account view to edit your account email address & password.
Change your email address and password to new values.
Save.
Try to log back in with the new values.

## Version of Vue Storefront
1.12.2

## Can you handle fixing this bug by yourself?
No

## Environment details

* Browser: Chrome v87.0.4280.88
* OS: Windows 10 Pro
* Node: v10.16.3
* Code Version: 1.12.2

### PR DESCRIPTION
**Pull request by Fifciu**
### Related Issues

closes #5315 

### Which Environment This Relates To
Check your case. In case of any doubts please read about [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html)

- [ ] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [ ] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [x] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog
- [ ] No upgrade steps required (100% backward compatibility and no breaking changes)
- [x] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/docs/guide/upgrade-notes/README.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing Vue Storefront sites with this new feature

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and Currently Important Rules Acceptance

- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: core/modules/user/components/UserAccount.ts [hunk 1]
```diff
@@ -144,16 +144,52 @@ export const UserAccount = {
...
+    changePasswordEmit () {
+      this.$bus.$emit('myAccount-before-changePassword', {
+        currentPassword: this.oldPassword,
+        newPassword: this.password
+      })
     },
-    exitSection (event, updatedProfile) {
-      this.$bus.$emit('myAccount-before-updateUser', updatedProfile)
+    exitSection (event, updatedProfile, changePassword) {
+      this.$bus.$emit('myAccount-before-updateUser', {
```
[the remark this answers is not in the document]
Comment (gibkigonzo) [reply core/modules/user/components/UserAccount.ts:161]: How about getting out of this madness and remove event? We need to make b/c anyway.
```
async exitSection (event, updatedProfile) {
        try {
          if (updatedProfile) {
            await this.$store.dispatch('user/update', { customer: updatedProfile })
          }
          // after updating user profile we can check if we need to update password
          // we don't care if 'updatedProfile' exist or not
          if (this.changePassword && this.password) {
            // we don't need to call it twice (lines 147-148 and 188-190)
            // with that change we will keep single responsibility in `updateProfile` (only prepare object) and in `exitSection` (makes all update calls)
            this.$bus.$emit('myAccount-before-changePassword', {
              currentPassword: this.oldPassword,
              newPassword: this.password
            })
          }
          // everything is ok then get out of edit mode ('myAccount-before-changePassword' can still fail and we will not handle that :D but we would need to remove it also)
          this.remainInEditMode = false
        } catch (err) {
          this.remainInEditMode = true
          // this.$bus.$emit('myAccount-before-remainInEditMode', this.$props.activeBlock) its the same as this.remainInEditMode = true
          throw new Error(err) // or push notification
        } finally {
            this.currentUser = Object.assign({}, this.$store.state.user.current)
            this.userCompany = this.getUserCompany()
            this.changePassword = false
            this.oldPassword = ''
            this.password = ''
            this.rPassword = ''
            this.addCompany = !!this.userCompany.company // I don't know what type it has, but for sure we don't need 'if' statement here
            // this.remainInEditMode = false; I think `if (!updatedProfile) {` was needed only for this line in other scenarios we always want to set current user (updateded or not)
            this.isEdited = this.remainInEditMode // we can even go further and remove 'remainInEditMode' and use only 'isEdited':D
        }
 ```
 With that change we will remove binding with `'myAccount-before-updateUser'` and `'myAccount-before-remainInEditMode'` You also don't need to create new method `changePasswordEmit`

File: core/modules/user/components/UserAccount.ts [hunk 2]
```diff
@@ -144,16 +144,52 @@ export const UserAccount = {
...
+              if (!this.userCompany.company) {
+                this.addCompany = false
+              }
+              this.remainInEditMode = false
+            }
+            if (!this.remainInEditMode) {
+              this.isEdited = false
+            }
+          } else {
+            this.$store.dispatch('notification/spawnNotification', {
```
[the remark this answers is not in the document]
Comment (Fifciu) [reply core/modules/user/components/UserAccount.ts:181]: But they would need to overwrite some core method anyway to add notification

File: docs/guide/upgrade-notes/README.md [hunk 3]
```diff
@@ -73,6 +73,14 @@ Another part you should keep in mind is Cache Key. If you render different compo
 
 I created dedicated hook to resolve this problem. You can find how to use it in `src/modules/device/README.md`. You also have commented example of usage in `src/modules/device/server.ts`.
 
+### core/data-resolver/UserService.ts
+`updateProfile` uses `TaskQueue.execute` instead of `TaskQueue.queue`
+
+### core/pages/MyAccount.js
+`onBeforeUpdateUser` changes signature from `(updatedProfile)` to `({ updatedProfile, callback })`. Callback is being called just after `user/update` actions sucesfully finishes.
```
Comment (gibkigonzo) [inline docs/guide/upgrade-notes/README.md:80]: `onBeforeUpdateUser` doesn't exist now :)

File: core/modules/user/components/UserShippingDetails.ts [hunk 4]
```diff
@@ -111,17 +105,29 @@ export const UserShippingDetails = {
           })
         }
       }
-      this.exitSection(null, updatedShippingDetails)
+      return this.exitSection(updatedShippingDetails)
     },
-    exitSection (event, updatedShippingDetails) {
```
Comment (gibkigonzo) [inline core/modules/user/components/UserShippingDetails.ts:116]: Please keep same api or add this change to upgrade notes

**Review by gibkigonzo — CHANGES_REQUESTED** [review]

**Review by gibkigonzo — APPROVED** [review]

**Pull request merged**